src/main/java/com/es/phoneshop/web/ProductDetailsPageServlet.java

```diff
@@ -1,25 +1,33 @@
 package com.es.phoneshop.web;
 
+import com.es.phoneshop.model.cart.Cart;
+import com.es.phoneshop.model.cart.CartItem;
+import com.es.phoneshop.model.cart.CartService;
+import com.es.phoneshop.model.cart.DefaultCartService;
+import com.es.phoneshop.model.exceptions.OutOfStockException;
 import com.es.phoneshop.model.exceptions.ProductNotFoundException;
-import com.es.phoneshop.model.product.ArrayListProductDao;
-import com.es.phoneshop.model.product.ProductDao;
-import com.es.phoneshop.model.product.SortField;
-import com.es.phoneshop.model.product.SortOrder;
+import com.es.phoneshop.model.product.*;
 
 import javax.servlet.ServletConfig;
 import javax.servlet.ServletException;
 import javax.servlet.http.HttpServlet;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
 import java.io.IOException;
+import java.text.NumberFormat;
+import java.text.ParseException;
 import java.util.Optional;
 
 public class ProductDetailsPageServlet extends HttpServlet {
 
     private ProductDao productDao;
+    private RecentlyViewedProducts recentlyViewedProducts;
+    private CartService cartService;
     @Override
     public void init(ServletConfig config) throws ServletException{
         super.init(config);
+        cartService= DefaultCartService.getInstance();
+        recentlyViewedProducts=RecentlyViewedProducts.getInstance();
         try {
             productDao=ArrayListProductDao.getInstance();
         } catch (ProductNotFoundException e) {
```

Comment:
You haven't search any product yet, why do you expect a ProductNotFoundException here?